### DISCUSSION TITLE
Is there a way to skip an expect if the previous expect fails?

### DISCUSSION THREAD

**Discussion by silasabbott** [vitest-dev/vitest#1170] [Q&A]
I have plenty of tests/expects that I know will fail, for example:
``` js
expect(store).toHaveProperty('ready')
expect(store.ready).toBeInstanceOf(Promise) // will always fail if the previous fails
```
When there's a whole string of expects that fail after one does, it throws a ton of fails in the console that are hard to sort through when it's only one actual problem.

Is this bad practice, or is there a way to do this that I'm missing?

**Comment by sheremet-va** [comment]
Are they in separate tests? If so, it is a bad practice. Tests should be isolated.

**Reply by silasabbott** [reply]
No, just a row of expects within a single test. The only reason I'm wondering is because when one expect fails, sometimes the reason it fails can cause many other expects to fail too. I'm trying to get vitest to show only the assertion error that matters so I don't have to scroll a bunch to find that a single variable was undefined or something like that
